rasterio/_warp.pyx

```diff
@@ -348,150 +352,138 @@ def _reproject(
     cdef void *hTransformArg = NULL
     cdef _gdal.GDALWarpOptions *psWOptions = NULL
     cdef GDALWarpOperation *oWarper = new GDALWarpOperation()
+    cdef int num_threads = int(kwargs.get('num_threads', 1))
     reprojected = False
 
-    try:
-        hTransformArg = _gdal.GDALCreateGenImgProjTransformer(
-                                            hdsin, NULL, hdsout, NULL, 
-                                            1, 1000.0, 0)
-        if hTransformArg == NULL:
-            raise ValueError("NULL transformer")
-        log.debug("Created transformer")
-
-        psWOptions = _gdal.GDALCreateWarpOptions()
-
-        # Note: warp_extras is pointed to different memory locations on every
-        # call to CSLSetNameValue call below, but needs to be set here to
-        # get the defaults
-        warp_extras = psWOptions.papszWarpOptions
-
-        for k, v in kwargs.items():
-            k, v = k.upper(), str(v).upper()
-            key_b = k.encode('utf-8')
-            val_b = v.encode('utf-8')
-            key_c = key_b
-            val_c = val_b
-            warp_extras = _gdal.CSLSetNameValue(warp_extras, key_c, val_c)
-            log.debug("Setting warp option  %s: %s" % (k, v))
-        
-        pszWarpThreads = _gdal.CSLFetchNameValue(warp_extras, "NUM_THREADS")
-        if pszWarpThreads == NULL:
-            pszWarpThreads = _gdal.CPLGetConfigOption(
-                                "GDAL_NUM_THREADS", "1")
-            warp_extras = _gdal.CSLSetNameValue(
-                warp_extras, "NUM_THREADS", pszWarpThreads)
-
-        log.debug("Created warp options")
+    hTransformArg = _gdal.GDALCreateGenImgProjTransformer(
+                                        hdsin, NULL, hdsout, NULL, 
+                                        1, 1000.0, 0)
+    if hTransformArg == NULL:
+        raise ValueError("NULL transformer")
+    log.debug("Created transformer")
+
+    psWOptions = _gdal.GDALCreateWarpOptions()
+
+    # Note: warp_extras is pointed to different memory locations on every
+    # call to CSLSetNameValue call below, but needs to be set here to
+    # get the defaults
+    warp_extras = psWOptions.papszWarpOptions
+
+    for k, v in kwargs.items():
+        k, v = k.upper(), str(v).upper()
+        key_b = k.encode('utf-8')
+        val_b = v.encode('utf-8')
+        key_c = key_b
+        val_c = val_b
+        warp_extras = _gdal.CSLSetNameValue(warp_extras, key_c, val_c)
+        log.debug("Setting warp option  %s: %s" % (k, v))
     
-        psWOptions.eResampleAlg = <_gdal.GDALResampleAlg>resampling
+    num_threads = kwargs.get('num_threads', 1)
 
-        # Set src_nodata and dst_nodata
-        if src_nodata is None and dst_nodata is not None:
-            raise ValueError("src_nodata must be provided because dst_nodata "
-                             "is not None")
-        log.debug("src_nodata: %s" % src_nodata)
+    #pszWarpThreads = _gdal.CSLFetchNameValue(warp_extras, "NUM_THREADS")
```

Comment:
@sgillies is the plan to reintroduce this at a later point?  Looks like the only usage of `num_threads` below is to switch between single and multiple threads.

If not, then can this block be removed?